crates/trie/sparse/src/trie.rs

```diff
@@ -665,30 +665,29 @@ impl RevealedSparseTrie {
                         continue
                     }
 
-                    branch_child_buf.clear();
                     // Walk children in a reverse order from `f` to `0`, so we pop the `0` first
                     // from the stack.
-                    for bit in CHILD_INDEX_RANGE.rev() {
+                    branch_child_buf.clear();
+                    for bit in CHILD_INDEX_RANGE {
                         if state_mask.is_bit_set(bit) {
                             let mut child = path.clone();
                             child.push_unchecked(bit);
                             branch_child_buf.push(child);
                         }
                     }
 
-                    branch_value_stack_buf.resize(branch_child_buf.len(), Default::default());
                     let mut added_children = false;
-                    for (i, child_path) in branch_child_buf.iter().enumerate() {
+                    for child_path in branch_child_buf.iter() {
                         if rlp_node_stack.last().map_or(false, |e| &e.0 == child_path) {
                             let (_, child) = rlp_node_stack.pop().unwrap();
                             // Insert children in the resulting buffer in a normal order, because
                             // initially we iterated in reverse.
-                            branch_value_stack_buf[branch_child_buf.len() - i - 1] = child;
+                            branch_value_stack_buf.insert(0, child);
```

Comment:
yeah, i did it that way at first, but this does a shift every time we insert, while `branch_value_stack_buf[index]` doesn't

Reply:
we know the length and can calculate the indices ahead of time, so IMO for a small perf gain we should rather do direct indexing